File: cfgs/java_cfgs_foo/REARRANGE_POSITIVE_AND_NEGATIVE_NUMBERS_PUBLISH.java
Nodes
1;special;;;
2;statement;int i = -1;14;
3;statement;int temp = 0;14;
4;statement;int j = 0;15;
5;conditional;foo_1 < foo_2;16;
6;conditional;foo_1[foo_2] < 0;18;
7;statement;i++;19;
8;statement;temp = foo_1[foo_2];20;
9;statement;foo_1[foo_2] = foo_3[foo_4];21;
10;statement;foo_1[foo_2] = foo_3;22;
11;statement;j++;17;
12;statement;int pos = foo_1 + 1;25;
13;statement;int neg = 0;25;
14;conditional;((foo_1 < foo_2) && (foo_3 < foo_4)) && (foo_5[foo_6] < 0);26;
15;statement;temp = foo_1[foo_2];27;
16;statement;foo_1[foo_2] = foo_3[foo_4];28;
17;statement;foo_1[foo_2] = foo_3;29;
18;statement;pos++;30;
19;statement;neg += 2;31;
20;exit;;;
Edges
0;2;
1;3;
2;4;
3;5;
4;6;12;
5;7;11;
6;8;
7;9;
8;10;
9;11;
10;5;
11;13;
12;14;
13;20;15;
14;16;
15;17;
16;18;
17;19;
18;14;
19;

File: cfgs/pythoncfgs/REARRANGE_POSITIVE_AND_NEGATIVE_NUMBERS_PUBLISH.py
Nodes
1;statement;foo_1 = -1;8;
2;conditional;fun_1(foo_1);9;
3;statement;(foo_1, foo_2) = (foo_3 + 1, 0);13;
4;statement;foo_1 < foo_2 and foo_3 < foo_4 and (foo_5[foo_6] < 0);14;
5;statement;(foo_1[foo_2], foo_3[foo_4]) = (foo_5[foo_6], foo_7[foo_8]);15;
6;statement;foo_1 += 1;16;
7;statement;foo_1 += 2;17;
8;conditional;foo_1[foo_2] < 0;10;
9;statement;foo_1 += 1;11;
10;statement;(foo_1[foo_2], foo_3[foo_4]) = (foo_5[foo_6], foo_7[foo_8]);12;
11;exit;;
Edges
0;2;
1;3;8;
2;4;
3;5;
4;6;
5;7;
6;4;
7;2;9;
8;10;
9;2;
10;
